software/runtime/runtime.h

```diff
@@ -48,6 +48,8 @@ static inline void mempool_init(const uint32_t core_id,
     alloc_init(get_alloc_l1(), &__heap_start, heap_size);
 
     // Initialize L1 Sequential Heap Allocator per Tile
+    // TODO: Hardcoded stuff? NUM_CORES_PER_TILE? number of words per queue
+    // entry?
```

Comment:
Should be something like this:

```suggestion
    extern int32_t __seq_start;
    uint32_t seq_heap_offset = NUM_CORES_PER_TILE * STACK_SIZE + NUM_BANKS_PER_TILE * sizeof(int32_t) * XQUEUE_SIZE;
    uint32_t seq_total_size = NUM_CORES_PER_TILE * SEQ_MEM_SIZE;
    int32_t *seq_heap_base = &__seq_start + (seq_heap_offset / sizeof(int32_t));
    uint32_t seq_heap_size = seq_total_size - seq_heap_offset;
    for (uint32_t tile_id = 0; tile_id < num_cores / NUM_CORES_PER_TILE; ++tile_id) {
      alloc_init(get_alloc_tile(tile_id), seq_heap_base, seq_heap_size);
      seq_heap_base += (seq_total_size / sizeof(int32_t));
    }
```